Potential timzone bug with logfile cleanup

Near the end of the day, I start to get errors in certain test cases. At the moment, I am only getting an error in the `query_logging_resolver_test` when attempting to clean up old log files. 

I believe this is due to the fact that `time.Parse` parses to UTC in the absence of timezone information in the string. I don't think it makes sense to add timezone information to the logs. Instead, let's just assume local timezone. This assumes that the user is running this on a server set to their local timezone or to UTC. In either case, this should work, as `time.Local` respects the system timezone. 

I'll submit a PR shortly after posting this issue.

For reference, I am located in the central US, currently at UTC-6:00 offset. 

<summary>
Test Logs
<details>
<pre>
➜  blocky-upstream git:(timezone-test-bug) make test
go run github.com/onsi/ginkgo/v2/ginkgo --label-filter="!e2e" --coverprofile=coverage.txt --covermode=atomic --cover -r -p
[1706929175] API Suite - 13/13 specs - 15 procs ••••••••••••• SUCCESS! 255.683482ms 
coverage: 11.6% of statements
[1706929175] Expiration cache suite - 19/19 specs - 15 procs ••••••••••••••••••• SUCCESS! 587.909369ms 
coverage: 100.0% of statements
[1706929175] String cache suite - 35/35 specs - 15 procs ••••••••••••••••••••••••••••••••••• SUCCESS! 363.974776ms 
coverage: 100.0% of statements
[1706929175] Command Suite - 29/29 specs - 15 procs •
------------------------------
• [0.001 seconds]
Version command when Version command is called should execute without error
/home/mchone/Documents/projects/blocky-upstream/cmd/version_test.go:10

  Captured StdOut/StdErr Output >>
  blocky
  Version: undefined
  Build time: undefined
  Architecture: undefined
  << Captured StdOut/StdErr Output
------------------------------
• [0.020 seconds]
Blocking command Call query command when Type is wrong should end with error
/home/mchone/Documents/projects/blocky-upstream/cmd/query_test.go:86

  Captured StdOut/StdErr Output >>
  Error: unknown query type 'X'
  Usage:
    query <domain> [flags]

  Flags:
    -h, --help          help for query
    -t, --type string   query type (A, AAAA, ...) (default "A")

  << Captured StdOut/StdErr Output
------------------------------
• [0.009 seconds]
Lists command Call list refresh command when list refresh is executed should print result
/home/mchone/Documents/projects/blocky-upstream/cmd/lists_test.go:43

  Captured StdOut/StdErr Output >>
  [2024-02-02 20:59:53]  INFO OK
  << Captured StdOut/StdErr Output
------------------------------
• [0.002 seconds]
Healthcheck command Call healthcheck command should fail
/home/mchone/Documents/projects/blocky-upstream/cmd/healthcheck_test.go:15

  Captured StdOut/StdErr Output >>
  NOT OK
  Error: dial tcp 127.0.0.1:533: connect: connection refused
  Usage:
    healthcheck [flags]

  Flags:
    -h, --help          help for healthcheck
    -p, --port uint16   healthcheck port 5333 (default 53)

  << Captured StdOut/StdErr Output
------------------------------
• [0.003 seconds]
Blocking command enable blocking when Enable blocking is called via REST should enable the blocking status
/home/mchone/Documents/projects/blocky-upstream/cmd/blocking_test.go:41

  Captured StdOut/StdErr Output >>
  [2024-02-02 20:59:53]  INFO OK
  << Captured StdOut/StdErr Output
------------------------------
•
------------------------------
• [0.008 seconds]
Lists command Call list refresh command when Server returns 500 should end with error
/home/mchone/Documents/projects/blocky-upstream/cmd/lists_test.go:58

  Captured StdOut/StdErr Output >>
  Error: response NOK, 500 Internal Server Error 
  Usage:
    refresh [flags]

  Flags:
    -h, --help   help for refresh

  << Captured StdOut/StdErr Output
------------------------------
••
------------------------------
• [0.007 seconds]
Lists command Call list refresh command when Url is wrong should end with error
/home/mchone/Documents/projects/blocky-upstream/cmd/lists_test.go:69

  Captured StdOut/StdErr Output >>
  Error: can't execute Post "http://127.0.0.1:0/api/lists/refresh": dial tcp 127.0.0.1:0: connect: connection refused
  Usage:
    refresh [flags]

  Flags:
    -h, --help   help for refresh

  << Captured StdOut/StdErr Output
------------------------------
••
------------------------------
• [0.004 seconds]
Blocking command disable blocking when disable blocking is called via REST should enable the blocking status
/home/mchone/Documents/projects/blocky-upstream/cmd/blocking_test.go:69

  Captured StdOut/StdErr Output >>
  [2024-02-02 20:59:53]  INFO OK
  << Captured StdOut/StdErr Output
------------------------------
•••
------------------------------
• [0.004 seconds]
root command when Config provided should accept new env var
/home/mchone/Documents/projects/blocky-upstream/cmd/root_test.go:61

  Captured StdOut/StdErr Output >>
  [2024-02-02 20:59:53]  WARN config option "port" is deprecated, please use "ports.dns" instead
  [2024-02-02 20:59:53] ERROR configuration uses deprecated options, see warning logs for details
  << Captured StdOut/StdErr Output
------------------------------
•
------------------------------
• [0.014 seconds]
Healthcheck command Call healthcheck command shoul succeed
/home/mchone/Documents/projects/blocky-upstream/cmd/healthcheck_test.go:24

  Captured StdOut/StdErr Output >>
  Mock helthcheck server is up
  NOT OK
  Error: dial tcp 127.0.0.1:5110: connect: connection refused
  Usage:
    healthcheck [flags]

  Flags:
    -h, --help          help for healthcheck
    -p, --port uint16   healthcheck port 5333 (default 53)

  OK
  << Captured StdOut/StdErr Output
------------------------------
•
------------------------------
• [0.004 seconds]
Cache command flush cache when flush cache is called via REST should flush caches
/home/mchone/Documents/projects/blocky-upstream/cmd/cache_test.go:37

  Captured StdOut/StdErr Output >>
  [2024-02-02 20:59:53]  INFO OK
  << Captured StdOut/StdErr Output
------------------------------
•••
------------------------------
• [0.009 seconds]
root command when Config provided should accept old env var
/home/mchone/Documents/projects/blocky-upstream/cmd/root_test.go:52

  Captured StdOut/StdErr Output >>
  [2024-02-02 20:59:53]  WARN config option "port" is deprecated, please use "ports.dns" instead
  [2024-02-02 20:59:53] ERROR configuration uses deprecated options, see warning logs for details
  << Captured StdOut/StdErr Output
------------------------------
• [0.052 seconds]
Serve command when Serve command is called with valid config should fail if server start fails
/home/mchone/Documents/projects/blocky-upstream/cmd/serve_test.go:74

  Captured StdOut/StdErr Output >>
  [2024-02-02 20:59:53]  INFO _/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/       _/        _/                      _/                   _/
  [2024-02-02 20:59:53]  INFO _/      _/_/_/    _/    _/_/      _/_/_/  _/  _/    _/    _/    _/
  [2024-02-02 20:59:53]  INFO _/     _/    _/  _/  _/    _/  _/        _/_/      _/    _/     _/
  [2024-02-02 20:59:53]  INFO _/    _/    _/  _/  _/    _/  _/        _/  _/    _/    _/      _/
  [2024-02-02 20:59:53]  INFO _/   _/_/_/    _/    _/_/      _/_/_/  _/    _/    _/_/_/       _/
  [2024-02-02 20:59:53]  INFO _/                                                    _/        _/
  [2024-02-02 20:59:53]  INFO _/                                               _/_/           _/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/  Version: undefined          Build time: undefined           _/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/
  [2024-02-02 20:59:53]  INFO bootstrap: bootstrapDns is not configured, will use system resolver
  [2024-02-02 20:59:53]  INFO server: current configuration:
  [2024-02-02 20:59:53]  INFO server: -> query_logging:
  [2024-02-02 20:59:53]  INFO server:      type: console
  [2024-02-02 20:59:53]  INFO server:      logRetentionDays: 0
  [2024-02-02 20:59:53]  INFO server:      flushInterval: 30 seconds
  [2024-02-02 20:59:53]  INFO server:      fields: [clientIP clientName responseReason responseAnswer question duration]
  [2024-02-02 20:59:53]  INFO server: -> caching:
  [2024-02-02 20:59:53]  INFO server:      minTime = 0 seconds
  [2024-02-02 20:59:53]  INFO server:      maxTime = 0 seconds
  [2024-02-02 20:59:53]  INFO server:      cacheTimeNegative = 30 minutes
  [2024-02-02 20:59:53]  INFO server:      cache entries = 0
  [2024-02-02 20:59:53]  INFO server: -> special_use_domains:
  [2024-02-02 20:59:53]  INFO server: -> parallel_best:
  [2024-02-02 20:59:53]  INFO server:      group: default
  [2024-02-02 20:59:53]  INFO server:      upstreams:
  [2024-02-02 20:59:53]  INFO server:        - tcp+udp:1.1.1.1
  [2024-02-02 20:59:53]  INFO server: listeners:
  [2024-02-02 20:59:53]  INFO server:   DNS   = [5001]
  [2024-02-02 20:59:53]  INFO server:   TLS   = []
  [2024-02-02 20:59:53]  INFO server:   HTTP  = []
  [2024-02-02 20:59:53]  INFO server:   HTTPS = []
  [2024-02-02 20:59:53]  INFO server: runtime information:
  [2024-02-02 20:59:53]  INFO server:   numCPU =       16
  [2024-02-02 20:59:53]  INFO server:   numGoroutine = 21
  [2024-02-02 20:59:53]  INFO server:   memory:
  [2024-02-02 20:59:53]  INFO server:     heap =              1 MB
  [2024-02-02 20:59:53]  INFO server:     sys =              14 MB
  [2024-02-02 20:59:53]  INFO server:     numGC =             3
  [2024-02-02 20:59:53]  INFO server: Starting server
  [2024-02-02 20:59:53]  INFO server: UDP server is up and running on address :5001
  [2024-02-02 20:59:53] ERROR server start failed: start tcp listener failed: listen tcp :5001: bind: address already in use
  << Captured StdOut/StdErr Output
------------------------------
•
------------------------------
• [0.063 seconds]
Serve command when Serve command is called with valid config should start without error and terminate with signal
/home/mchone/Documents/projects/blocky-upstream/cmd/serve_test.go:32

  Captured StdOut/StdErr Output >>
  [2024-02-02 20:59:53]  INFO _/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/       _/        _/                      _/                   _/
  [2024-02-02 20:59:53]  INFO _/      _/_/_/    _/    _/_/      _/_/_/  _/  _/    _/    _/    _/
  [2024-02-02 20:59:53]  INFO _/     _/    _/  _/  _/    _/  _/        _/_/      _/    _/     _/
  [2024-02-02 20:59:53]  INFO _/    _/    _/  _/  _/    _/  _/        _/  _/    _/    _/      _/
  [2024-02-02 20:59:53]  INFO _/   _/_/_/    _/    _/_/      _/_/_/  _/    _/    _/_/_/       _/
  [2024-02-02 20:59:53]  INFO _/                                                    _/        _/
  [2024-02-02 20:59:53]  INFO _/                                               _/_/           _/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/  Version: undefined          Build time: undefined           _/
  [2024-02-02 20:59:53]  INFO _/                                                              _/
  [2024-02-02 20:59:53]  INFO _/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/
  [2024-02-02 20:59:53]  INFO bootstrap: bootstrapDns is not configured, will use system resolver
  [2024-02-02 20:59:53]  INFO server: current configuration:
  [2024-02-02 20:59:53]  INFO server: -> query_logging:
  [2024-02-02 20:59:53]  INFO server:      type: console
  [2024-02-02 20:59:53]  INFO server:      logRetentionDays: 0
  [2024-02-02 20:59:53]  INFO server:      flushInterval: 30 seconds
  [2024-02-02 20:59:53]  INFO server:      fields: [clientIP clientName responseReason responseAnswer question duration]
  [2024-02-02 20:59:53]  INFO server: -> caching:
  [2024-02-02 20:59:53]  INFO server:      minTime = 0 seconds
  [2024-02-02 20:59:53]  INFO server:      maxTime = 0 seconds
  [2024-02-02 20:59:53]  INFO server:      cacheTimeNegative = 30 minutes
  [2024-02-02 20:59:53]  INFO server:      cache entries = 0
  [2024-02-02 20:59:53]  INFO server: -> special_use_domains:
  [2024-02-02 20:59:53]  INFO server: -> parallel_best:
  [2024-02-02 20:59:53]  INFO server:      group: default
  [2024-02-02 20:59:53]  INFO server:      upstreams:
  [2024-02-02 20:59:53]  INFO server:        - tcp+udp:1.1.1.1
  [2024-02-02 20:59:53]  INFO server: listeners:
  [2024-02-02 20:59:53]  INFO server:   DNS   = [5002]
  [2024-02-02 20:59:53]  INFO server:   TLS   = []
  [2024-02-02 20:59:53]  INFO server:   HTTP  = []
  [2024-02-02 20:59:53]  INFO server:   HTTPS = []
  [2024-02-02 20:59:53]  INFO server: runtime information:
  [2024-02-02 20:59:53]  INFO server:   numCPU =       16
  [2024-02-02 20:59:53]  INFO server:   numGoroutine = 20
  [2024-02-02 20:59:53]  INFO server:   memory:
  [2024-02-02 20:59:53]  INFO server:     heap =              1 MB
  [2024-02-02 20:59:53]  INFO server:     sys =              14 MB
  [2024-02-02 20:59:53]  INFO server:     numGC =             3
  [2024-02-02 20:59:53]  INFO server: Starting server
  [2024-02-02 20:59:53]  INFO server: TCP server is up and running on address :5002
  [2024-02-02 20:59:53]  INFO server: UDP server is up and running on address :5002
  [2024-02-02 20:59:53]  INFO Terminating...
  [2024-02-02 20:59:53]  INFO server: Stopping server
  << Captured StdOut/StdErr Output
------------------------------
 SUCCESS! 79.77309ms 
coverage: 89.1% of statements
[1706929175] Config Suite - 182/182 specs - 15 procs •••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• SUCCESS! 215.317021ms 
coverage: 80.8% of statements
[1706929175] e2e Suite [e2e] - 0/22 specs - 15 procs SSSSSSSSSSSSSSSSSSSSSS SUCCESS! 41.615344ms 
coverage: [no statements]
[1706929175] Lists Suite - 25/25 specs - 15 procs ••••••••••••••••••••••••• SUCCESS! 1.035827122s 
coverage: 85.6% of statements
[1706929175] Parsers Suite - 35/35 specs - 15 procs ••••••••••••••••••••••••••••••••••• SUCCESS! 48.493104ms 
coverage: 100.0% of statements
[1706929175] Querylog Suite - 19/19 specs - 15 procs ••••••••••••••••••• SUCCESS! 204.761945ms 
coverage: 100.0% of statements
[1706929175] Redis Suite - 13/13 specs - 15 procs ••••••••••••• SUCCESS! 2.037535162s 
coverage: 90.3% of statements
[1706929175] Resolver Suite - 352/352 specs - 15 procs ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••
------------------------------
• [FAILED] [1.003 seconds]
QueryLoggingResolver Clean up of query log directory when log directory contains old files [It] should remove files older than defined log retention
/home/mchone/Documents/projects/blocky-upstream/resolver/query_logging_resolver_test.go:334

  [FAILED] Timed out after 1.001s.
  The function passed to Eventually failed at /home/mchone/Documents/projects/blocky-upstream/resolver/query_logging_resolver_test.go:345 with:
  Expected
      <string>: /tmp/queryLoggingResolver3224634037/2024-01-26-test.log
  to exist
  In [It] at: /home/mchone/Documents/projects/blocky-upstream/resolver/query_logging_resolver_test.go:347 @ 02/02/24 20:59:59.507
------------------------------
•••

Summarizing 1 Failure:
  [FAIL] QueryLoggingResolver Clean up of query log directory when log directory contains old files [It] should remove files older than defined log retention
  /home/mchone/Documents/projects/blocky-upstream/resolver/query_logging_resolver_test.go:347

Ran 352 of 352 Specs in 12.252 seconds
FAIL! -- 351 Passed | 1 Failed | 0 Pending | 0 Skipped

coverage: 97.2% of statements
composite coverage: 77.5% of statements

Ginkgo ran 14 suites in 35.051299079s

There were failures detected in the following suites:
  resolver ./resolver

Test Suite Failed
exit status 1
make: *** [Makefile:60: test] Error 1
</pre>
</details>
</summary>

--- a/querylog/file_writer.go
+++ b/querylog/file_writer.go
@@ -85,7 +85,7 @@ func (d *FileWriter) CleanUp() {
 	// search for log files, which names starts with date
 	for _, f := range files {
 		if strings.HasSuffix(f.Name(), ".log") && len(f.Name()) > 10 {
-			t, err := time.Parse("2006-01-02", f.Name()[:10])
+			t, err := time.ParseInLocation("2006-01-02", f.Name()[:10], time.Local)
 			if err == nil {
 				differenceDays := uint64(time.Since(t).Hours() / hoursPerDay)
 				if d.logRetentionDays > 0 && differenceDays > d.logRetentionDays {

